clients/vscode/src/chat/webview.ts

```diff
@@ -456,6 +462,53 @@ export class ChatWebview {
         const fileContext = await getFileContextFromSelection(editor, this.gitProvider);
         return fileContext;
       },
+      listFileInWorkspace: async (params: ListFilesInWorkspaceParams): Promise<ListFileItem[]> => {
+        const maxResults = params.limit || 50;
+        const searchQuery = params.query?.trim();
+
+        if (!searchQuery) {
+          const openTabs = window.tabGroups.all
+            .flatMap((group) => group.tabs)
+            .filter((tab) => tab.input && (tab.input as TabInputText).uri);
+
+          this.logger.info(`No query provided, listing ${openTabs.length} opened editors.`);
+          return openTabs.map((tab) => uriToListFileItem((tab.input as TabInputText).uri, this.gitProvider));
+        }
+
+        try {
+          const caseInsensitivePattern = searchQuery
+            .split("")
+            .map((char) => {
+              if (char.toLowerCase() !== char.toUpperCase()) {
+                return `{${char.toLowerCase()},${char.toUpperCase()}}`;
+              }
+              return escapeGlobPattern(char);
+            })
+            .join("");
+
+          const globPattern = `**/${caseInsensitivePattern}*`;
+
+          this.logger.info(`Searching files with pattern: ${globPattern}, limit: ${maxResults}`);
+
+          const files = await workspace.findFiles(globPattern, null, maxResults);
```

Comment:
Should we set exclude to `undefined` to use default file-excludes?
```suggestion
          const files = await workspace.findFiles(globPattern, undefined, maxResults);
```

Reply:
cc @liangfung